### ISSUE TITLE
Create an Ocean-Specific Implementation of libp2p

### ISSUE THREAD

**Issue by bogdanfazakas** [oceanprotocol/ocean-p2p-lib-js#1]
Develop an Ocean-specific implementation of the libp2p library, tailored to meet the needs of the Ocean ecosystem:

1.   Exported Objects:
•	The module should export two distinct objects:
       •	Server Object: Designed for ocean nodes.
       •	Client Object: Intended for workers and distributors in the nodes monitoring service and for the Ocean.js SDK.
2.	Initialisation and Configuration:
•	Each object must accept a configuration parameter at initialisation.
•	Configuration must differ between the server and client implementations, including but not limited to:
       •	Server-Specific Config: Private keys, timeout settings, and additional configurations required for running as an ocean node.
       •	Client-Specific Config: Lightweight configuration for connecting to ocean nodes or performing worker/distributor tasks.
3.	Use Cases:
       •	The Server Object will serve as the backbone for ocean nodes, providing network capabilities for communication and data exchange.
       •	The Client Object will facilitate:
               •	Operations for workers and distributors in the nodes monitoring service.
              •	Interactions with ocean nodes via the Ocean.js SDK.

**Comment by jamiehewitt15** [comment]
I thought we were going to do this in ocean.js? Or is the plan now to have a separate library for this?

**Comment by bogdanfazakas** [comment]
No, we'll have a separate to deal with p2p and that will be imported in ocean.js as well, we'll be using the client class for that use case here

**Comment by jamiehewitt15** [comment]
> No, we'll have a separate to deal with p2p and that will be imported in ocean.js as well, we'll be using the client class for that use case here

Ok cool, makes sense

**Comment by giancu4** [comment]
Explored, decided not to implement.
